skipchain/struct.go

```diff
@@ -535,3 +627,64 @@ func (sbm *SkipBlockMap) GetFuzzy(id string) *SkipBlock {
 	}
 	return nil
 }
+
+// store stores the skipblock into the database.
+// an error is returned on failure.
+func (db *SkipBlockDB) store(sb *SkipBlock) error {
+	return db.Update(func(tx *bolt.Tx) error {
+		key := sb.Hash
+		val, err := network.Marshal(sb)
+		if err != nil {
+			return err
+		}
+
+		return tx.Bucket([]byte(db.bucketName)).Put(key, val)
+	})
+}
+
+// get returns the skipblock identified by sbID.
+// nil is returned if the key does not exist.
+// An error is thrown if marshalling fails.
+func (db *SkipBlockDB) get(sbID SkipBlockID) (*SkipBlock, error) {
+	var sb *SkipBlock
+	err := db.View(func(tx *bolt.Tx) error {
+		val := tx.Bucket([]byte(db.bucketName)).Get(sbID)
+		if val == nil {
+			sb = nil
+			return nil
+		}
+
+		_, sbMsg, err := network.Unmarshal(val, cothority.Suite)
+		if err != nil {
+			return err
+		}
+
+		sb = sbMsg.(*SkipBlock).Copy()
+		return nil
+	})
+
+	return sb, err
+}
+
+// dump returns all the data in the database as a map
```

Comment:
This feels scary if we are talking about a multi-gigabyte database. Should there be a safety limit on it, telling the caller, "errors.New("result would be too big")"

Not very nice for the caller, but better than blowing up RAM, crashing, etc.

Reply:
All the dump related methods are going to be removed after @ineiti finishes his skipchain PR. What do you think @ineti, should we add some limit on this now or remove it after we merge?

Reply:
I'd remove it after merge. The `GetAllSkipchains` method to the service needs to be removed and replaced by a `GetSkipchainUpdate` where you can give a maximum range of skipchains to be returned.